packages/server/graphql/mutations/addMissingJiraField.ts

```diff
@@ -0,0 +1,116 @@
+import {GraphQLID, GraphQLNonNull} from 'graphql'
+import {getUserId, isTeamMember} from '../../utils/authorization'
+import AddMissingJiraFieldPayload from '../types/AddMissingJiraFieldPayload'
+import {GQLContext} from '../graphql'
+import AtlassianServerManager from '../../utils/AtlassianServerManager'
+import isPhaseComplete from '~/utils/meetings/isPhaseComplete'
+import MeetingPoker from '../../database/types/MeetingPoker'
+import EstimatePhase from '../../database/types/EstimatePhase'
+import getTemplateRefById from '../../postgres/queries/getTemplateRefById'
+import {TaskServiceEnum} from '../../database/types/Task'
+import JiraServiceTaskId from '~/shared/gqlIds/JiraServiceTaskId'
+import publish from '../../utils/publish'
+import {SubscriptionChannel} from '~/types/constEnums'
+
+const addMissingJiraField = {
+  type: GraphQLNonNull(AddMissingJiraFieldPayload),
+  description: `Adds a missing Jira field to a screen currently assigned to a Jira project`,
+  args: {
+    meetingId: {
+      type: GraphQLNonNull(GraphQLID)
+    },
+    stageId: {
+      type: GraphQLNonNull(GraphQLID)
+    }
+  },
+  resolve: async (
+    _source,
+    {meetingId, stageId},
+    {authToken, dataLoader, socketId: mutatorId}: GQLContext
+  ) => {
+    const viewerId = getUserId(authToken)
+    const operationId = dataLoader.share()
+    const subOptions = {mutatorId, operationId}
+
+    //AUTH
+    const meeting = (await dataLoader.get('newMeetings').load(meetingId)) as MeetingPoker
+    if (!meeting) {
+      return {error: {message: 'Meeting not found'}}
+    }
+    const {
+      endedAt,
+      phases,
+      meetingType,
+      teamId,
+      createdBy,
+      facilitatorUserId,
+      templateRefId
+    } = meeting
+    if (!isTeamMember(authToken, teamId)) {
+      return {error: {message: 'Not on the team'}}
+    }
+    if (endedAt) {
+      return {error: {message: 'Meeting has ended'}}
+    }
+    if (meetingType !== 'poker') {
+      return {error: {message: 'Not a poker meeting'}}
+    }
+    if (isPhaseComplete('ESTIMATE', phases)) {
```

Comment:
Nit: if I pass in the `stageId` for the SCOPE phase, the error message seems confusing as I haven't started the ESTIMATE phase, how can it already be completed?

Reply:
I'm not sure about the exact flow but I took the validation steps from `pokerSetFinalScore` which I believe should be similar. I'm happy to improve if that's not needed though.